### PR TITLE
[MLDB-2163] add support to call functions and query with POST

### PR DESCRIPTION
**Pull request by guyd**

### CODE REVIEW

**Comment by guyd** [comment]
@FinchPowers What is missing?

**Comment by FinchPowers** [comment]
* routes /v1/query and /v1/functions must not to be modified.
* route /v1/get must be added.

File: server/mldb_server.cc [hunk 1]
```diff
@@ -217,6 +217,14 @@ initRoutes()
                                      "Do we sort the column names",
                                      false));
 
+        addRouteAsync(
+            versionNode, "/redirect/get", {"POST"}, "Redirect POST as GET with body. "
+            "Use this route only with system that does not support sending a GET with a body.",
```
Comment (FinchPowers) [inline server/mldb_server.cc]: system>s that do not<

File: server/mldb_server.cc [hunk 2]
```diff
@@ -263,6 +271,32 @@ runHttpQuery(const Utf8String& query,
                        rowNames, rowHashes, sortColumns);
 }
 
+void
+MldbServer::
+handleRedirectToGet(const string & uri,
+                    RestConnection & connection,
+                    const Json::Value & body) const
+{
```
Comment (FinchPowers) [inline server/mldb_server.cc]: Is there any risk that someone could make a call to this route to get something not related to mldb?
Say I am on a LAN. I have no access to the extranet but I have access to MLDB. MLDB has access to the extranet. Can I use MLDB to get out?

This is a security risk question.

Comment (guyd) [inline server/mldb_server.cc]: Yes.  That seems possible.

Comment (FinchPowers) [inline server/mldb_server.cc]: Would checking that the target starts with `/v1` be enough to block such attempts?

Comment (jeremybarnes) [inline server/mldb_server.cc]: It's not possible.  We don't actually make an HTTP call; we simply call the handleRequest method.  So there is no security issue here.

**Comment by jeremybarnes** [comment]
+1 from me

**Pull request merged**